docs/middleware.md

```diff
@@ -0,0 +1,76 @@
+# Using Next 12 middleware on Netlify
+
+Next 12 introduces a new feature called [middleware](https://nextjs.org/docs/middleware), which are functions that are
+run before a request has finished processing, and can be used to modify the request or replace the response. For
+example, it can change headers, rewrite the request path, or return a different response entirely.
+
+Next Middleware can run either in an edge function or at the origin. On Netlify, middleware runs at the origin as part
```

Comment:
Some links to `edge function` and `origin` would be useful

Reply:
Reading it back, I think it may need to be clearer that you can't run it in an edge function on Netlify.

Reply:
Minor suggestion for clarity:

```suggestion
Next.js Middleware can run either in an edge function or at the origin. On Netlify, middleware runs at the origin as part
```